sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1340,10 +1340,17 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
         val name = conf.getConfString("hive.script.serde",
           "org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe")
         val props = Seq("field.delim" -> "\t")
-        val recordHandler = Try(conf.getConfString(configKey)).toOption
+        val recordHandler = defaultRecordHandler(configKey)
         (Nil, Option(name), props, recordHandler)
     }
 
+    def defaultRecordHandler(configKey: String): Option[String] = {
+      Try(conf.getConfString(configKey)).orElse(Try(configKey match {
+        case "hive.script.recordreader" => "org.apache.hadoop.hive.ql.exec.TextRecordReader"
+        case "hive.script.recordwriter" => "org.apache.hadoop.hive.ql.exec.TextRecordWriter"
+      })).toOption
+    }
+
     val (inFormat, inSerdeClass, inSerdeProps, reader) =
       format(inRowFormat, "hive.script.recordreader")
```

Comment:
Can we pass in the default value for the reader/writer? like `format(inRowFormat, "hive.script.recordreader", "org.apache.hadoop.hive.ql.exec.TextRecordReader")` and `format(outRowFormat, "hive.script.recordwriter", "org.apache.hadoop.hive.ql.exec.TextRecordWriter")`. Then, in `def format`, we just use `getConfString(key: String, defaultValue: String)`